libp2p/dial.nim

```diff
@@ -17,6 +17,10 @@ import peerid,
 type
   Dial* = ref object of RootObj
 
+  DialRes* = tuple
```

Comment:
What is the reason to return the dialed protocol?

Reply:
For this v2 version, I merged the relayv1 and the relayv2, and they use two (three actually, but let's say two) different protocols. So I need to know what protocol is used when I dial in the [relay transport](https://github.com/lchenut/nim-libp2p/blob/566223a05ca0f30c1b0801df3748ec3c719787cf/libp2p/protocols/relay/rtransport.nim#L67-L78) for example.